packages/ec-core/src/distributions/conversion.rs

```diff
@@ -120,44 +283,54 @@ where
 {
     type Distribution = ChooseCloning<'a, U>;
 
-    type Error = EmptySlice;
+    type Error = Empty;
 
+    /// Creates a [`Distribution`] which uniformly samples a single element from
+    /// a [`&[T; N]`](primitive@array) and clones it.
+    ///
+    /// # Example
+    /// ```
+    /// # use ec_core::distributions::conversion::IntoDistribution;
+    /// # use rand::{rng, distr::{Distribution, slice::Empty}};
+    /// #
+    /// # fn main() -> Result<(),Empty> {
+    /// let my_array: [i32; 3] = [0, 1, 2];
+    /// let my_distr = IntoDistribution::<i32>::into_distribution(&my_array)?;
+    ///
+    /// let my_value: i32 = my_distr.sample(&mut rng());
+    /// # let _ = my_value;
+    /// # Ok(())
+    /// # }
+    /// ```
     fn into_distribution(self) -> Result<Self::Distribution, Self::Error> {
-        ToDistribution::<U>::to_distribution(self)
-    }
-}
-
-impl<'a, U, const N: usize> ToDistribution<'a, U> for [U; N]
-where
-    U: 'a + Clone,
-{
-    type Distribution = ChooseCloning<'a, U>;
-    type Error = EmptySlice;
-
-    fn to_distribution(&'a self) -> Result<Self::Distribution, Self::Error> {
         ChooseCloning::new(self)
     }
 }
 
-impl<'a, U, const N: usize> ToDistribution<'a, &'a U> for [U; N]
-where
-    U: 'a,
-{
-    type Distribution = Choose<'a, U>;
-    type Error = EmptySlice;
-
-    fn to_distribution(&'a self) -> Result<Self::Distribution, Self::Error> {
-        Choose::new(self).map_err(|_| EmptySlice)
-    }
-}
-
 impl<'a, T> IntoDistribution<&'a T> for &'a [T] {
     type Distribution = Choose<'a, T>;
 
-    type Error = EmptySlice;
+    type Error = Empty;
 
+    /// Creates a [`Distribution`] which uniformly samples a single element from
+    /// a [`&[U]`](primitive@slice) and returns a reference to it.
```

Comment:
Is there a typo here?

```suggestion
    /// a [`&[T]`](primitive@slice) and returns a reference to it.
```